cli/IOInterface.hpp

```diff
@@ -0,0 +1,62 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ **/
+
+#ifndef QUICKSTEP_CLI_IO_INTERFACE_HPP_
+#define QUICKSTEP_CLI_IO_INTERFACE_HPP_
+
+#include <string>
+
+/**
+ * Virtual base defines a generic, file-based interface around IO.
+ */
+class IOInterface {
+ public:
+  IOInterface() {}
+
```

Comment:
I removed the constructor but ran into a compiler error:
```
/Users/cramja/workspace/quickstep/incubator-quickstep/cli/NetworkIO.cpp:50:12: error: constructor for 'quickstep::NetworkIO' must explicitly initialize the base class 'quickstep::IOInterface' which does not have a default constructor
NetworkIO::NetworkIO() {
           ^
/Users/cramja/workspace/quickstep/incubator-quickstep/cli/IOInterface.hpp:62:7: note: 'quickstep::IOInterface' declared here
class IOInterface {
      ^
```
On doing the suggestion in the output, that too gave a compiler error.